File: 169-majority-element/majority-element.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        
        // Count occurrences
        for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        
        int majority = 0;
        int maxCount = 0;
        
        // Find element with maximum count
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() > maxCount) {
                majority = entry.getKey();
                maxCount = entry.getValue();
            }
        }
        
        return majority;
    }
}

File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        dict={}
        for i in nums:
            if i not in dict.keys():
                dict[i]=1
            else:
                dict[i]+=1
        pair=[0,0]
        for key,value in dict.items():
            if value>pair[1]:
                pair[0]=key
                pair[1]=value
        return pair[0]

        
        